compiler/rustc_errors/src/lib.rs

```diff
@@ -283,20 +283,25 @@ impl CodeSuggestion {
                 let mut buf = String::new();
 
                 let mut line_highlight = vec![];
+                // We need to keep track of the difference between the existing code and the added
+                // or deleted code in order to point at the correct column *after* substitution.
+                let mut acc = 0;
```

Comment:
Do you have an example of a problem that's fixed by this?

Reply:
It's for cases where there are multiple replacements in the same line, like in the following:

With acc:
![Screen Shot 2021-08-17 at 8 09 20 AM](https://user-images.githubusercontent.com/1606434/129752083-8b10f4bc-ba4f-4d54-9dc6-191a00c5d426.png)

Without acc:
![Screen Shot 2021-08-17 at 8 09 30 AM](https://user-images.githubusercontent.com/1606434/129752125-35799680-291e-421b-a303-29f22d7d077f.png)

It accounts for the shift in position of the later spans caused by the earlier replacements.

But it's _still_ buggy on the current commit in the presence of unicode:

With acc:
![Screen Shot 2021-08-17 at 8 09 07 AM](https://user-images.githubusercontent.com/1606434/129752215-ec3137e5-74fd-418c-9540-560d1d8f2fc6.png)

Without acc:
![Screen Shot 2021-08-17 at 8 08 57 AM](https://user-images.githubusercontent.com/1606434/129752272-d5258a6c-707c-4911-8e9a-280d262da954.png)

I'm cleaning up and pushing a new change that as far as I can see handles anything I throw at it correctly (the `%` are everything that is supposed to be highlighted and there only for debugging):
![Uploading Screen Shot 2021-08-17 at 8.44.07 AM.png…]()

Reply:
Also, @m-ou-se, it might be useful to check the PR on your sampling of terminals, but I expect things to be "fine" :)

Reply:
It's something I'd like to add at some point but not as part of this PR. It should be easy and self-contained, as a `-Z` flag, but I won't be having the time to do it in the short term.